### ISSUE TITLE
counterfact_grpo crashes on unsloth backend: tensor shape mismatch (1024 vs 256, dim 2) in training; works on trl

### ISSUE THREAD

**Issue by mykola-lexsi** [Lexsi-Labs/aligntune#20]
### Summary

`counterfact_grpo` training **crashes 100% of the time on the unsloth backend** with a tensor-shape mismatch, while the identical configuration trains fine with `backend="trl"`. Hit while benchmarking agent-generated code against aligntune — every counterfactual-GRPO task that let the backend default to unsloth failed at `.train()`.

### Error

```
[aligntune] ERROR aligntune.backends.unsloth.rl.counterfact_grpo.counterfact_grpo:872 -
Counterfactual GRPO training failed: The size of tensor a (1024) must match the size of tensor b (256) at non-singleton dimension 2
```

The trainer object is created without complaint; the crash is inside training, and the shape pair (1024 vs 256 at dim 2) is stable across runs/tasks — it looks like two different sequence/completion-length configs meeting in the counterfactual-logits path, not input-data noise.

### Repro

```python
from aligntune.core.factory import create_rl_trainer  # aligntune==0.1.7

trainer = create_rl_trainer(
    algorithm="counterfact_grpo",
    model_name="Qwen/Qwen2.5-0.5B",          # small model; same failure with Qwen3-0.6B
    dataset_name="tatsu-lab/alpaca",
    num_generations=2,
    batch_size=2,
    # backend defaults to unsloth when available -> crash
)
result = trainer.train()   # -> "size of tensor a (1024) must match ... b (256)"
```

Passing `backend="trl"` with otherwise identical arguments trains to completion.

### Environment

- aligntune 0.1.7 (pip), Python 3.12.13, torch 2.7–2.8 +cu128, single L40S
- Observed across ~10 benchmark tasks and multiple runs on our side — deterministic on unsloth, zero failures on trl

### Workaround we're shipping

We pin `backend="trl"` for `counterfact_grpo` in our task grounding, so no urgency on our side — filing so the unsloth path gets a look (or, if counterfact_grpo is known-unsupported on unsloth, a clean `NotImplementedError`/fallback at factory time would beat a mid-training shape crash).
